File: 01-GitHub_Copilot_presentation/JavaScript/math_operations.js
/**
 * Returns the sum of two numbers.
 *
 * @param {number} a - The first addend.
 * @param {number} b - The second addend.
 * @returns {number} the sum of the two numbers.
 */
function add(a, b) {
	return a + b;
}

/**
 * Returns the product of two numbers.
 *
 * @param {number} a - The multiplicand.
 * @param {number} b - The multiplier.
 * @returns {number} the product of the two numbers.
 */
function multiply(a, b) {
	return a * b;
}

/**
 * Returns the division of a by b.
 *
 * @param {number} a - The dividend.
 * @param {number} b - The divisor.
 * @throws {Error} If b is zero.
 * @returns {number} the division result.
 */
function divide(a, b) {
	if (b === 0)
		throw new Error("Cannot divide by zero");
	return a / b;
}


module.exports = { add, multiply, divide };

File: 01-GitHub_Copilot_presentation/Python/math_operations.py
def add(a: float, b: float) -> float:
	"""Returns the sum of two numbers.

	Parameters
	----------
		a: float

			The first addend.

		b: float

			The second addend.

		Returns
		-------
			float

				The addition of the two numbers.
	"""
	return a + b

def multiply(a: float, b: float) -> float:
	"""Returns the product of two numbers.

	Parameters
	----------
		a: float

			The multiplicand.

		b: float

			The multiplier.

		Returns
		-------
			float

				The product of the two numbers.
	"""
	return a * b

def divide(a: float, b: float) -> float:
	""" Returns the division of a by b.

	Parameters
	----------

		a : float
			The dividend.

		b : float
			The divisor.

	Raises
	------
		ValueError

			If b is zero.

	Returns
	-------
		float
			The division of the two numbers.
	"""
	if b == 0:
		raise ValueError("Cannot divide by zero")
	return a / b
